src/WebJobs.Extensions.DurableTask.Analyzers/Analyzers/Entity/DispatchClassNameAnalyzer.cs

```diff
@@ -44,26 +42,15 @@ private void AnalyzeDispatchEntityName(SyntaxNodeAnalysisContext context)
                 {
                     if (TryGetTypeArgumentList(expression, out SyntaxNode typeArgumentList))
                     {
-                        if(!TryGetIdentifierName(out SyntaxNode identifierName, typeArgumentList))
+                        if (SyntaxNodeUtils.TryGetClassSymbol(expression, context.SemanticModel, out INamedTypeSymbol classSymbol))
                         {
-                            var diagnostic = Diagnostic.Create(IncorrectTypeRule, typeArgumentList.GetLocation(), typeArgumentList);
-
-                            context.ReportDiagnostic(diagnostic);
-
-                            return;
-                        }
-                        else
-                        {
-                            if (SyntaxNodeUtils.TryGetClassSymbol(expression, context.SemanticModel, out INamedTypeSymbol classSymbol))
+                            var className = classSymbol.Name.ToString();
+                            var identifierName = typeArgumentList.ChildNodes().First();
```

Comment:
`First()` will throw an exception if `ChildNodes()` is an empty list. Do we know that `typeArgumentList.GetChildNodes()` will always return at least one identifier?

Reply:
TypeArgumentList will always have a child node. If the type is omitted its called OmittedTypeArgument. I will add a comment to this.